src/persist/src/indexed/mod.rs

```diff
@@ -334,8 +335,28 @@ impl<U: Buffer, L: Blob> Indexed<U, L> {
     /// smarts about waiting to call it only after there have been some writes),
     /// but it's exposed this way so we can write deterministic tests.
     pub fn step(&mut self) -> Result<(), Error> {
-        self.drain_future()
-        // TODO: Incrementally compact future.
+        self.drain_future()?;
+        let compaction_start = Instant::now();
+        let prev = self.serialize_meta();
+        for (_, future) in self.futures.iter_mut() {
+            if let Err(e) = future.step(&mut self.blob) {
```

Comment:
let's pull this whole bit into a method that returns Result so we can have the Err/restore handling in one place for this

Reply:
done!